samples/cpp/fitellipse.cpp

```diff
@@ -276,39 +276,34 @@ void processImage(int /*h*/, void*)
     {
         vector<Point2f> pts = points[i];
 
-        if (pts.size()<=5) {
+        //At least 5 points can fit an ellipse
+        if (pts.size()<5) {
             continue;
         }
         if (fitEllipseQ) {
             box = fitEllipse(pts);
-            if( MAX(box.size.width, box.size.height) > MIN(box.size.width, box.size.height)*30 ||
-               MAX(box.size.width, box.size.height) <= 0 ||
-               MIN(box.size.width, box.size.height) <= 0){continue;};
+            //size.height >= size.width awalys,only if the pts are on a line or at the same point,size.width=0
+            if (box.size.height <= box.size.width * 30 &&
```

Comment:
It would be better to extract repeating expression to a function:
```.cpp
inline static bool isGoodBox(const RotatedRect & box) { 
    return (box.size.height <= box.size.width * 30) && (box.size.width > 0); 
}
...
if (isGoodBox(box)) // draw
...
if (isGoodBox(boxAMS)) // draw
...
```
Or move this check to the `drawEllipseWithBox` method.